### ISSUE TITLE
Question about training loss behavior on small DL3DV subsets

### ISSUE THREAD

**Issue by uJunLI** [OpenRobotLab/AnySplat#50]
I just wanted to do a quick test, so I randomly selected three scenes from DL3DV for training and testing. However, I noticed that the loss did not decrease during training. Is this expected behavior for such a small subset, or is it necessary to use a larger-scale dataset in order for the loss to decrease?

**Comment by jianglh-WHU** [comment]
To be honest, I’m not sure. I haven’t conducted experiments where the model is overfitted on a small subset. You might need to try using a larger one.

**Comment by uJunLI** [comment]
> To be honest, I’m not sure. I haven’t conducted experiments where the model is overfitted on a small subset. You might need to try using a larger one.

<img width="1853" height="960" alt="Image" src="https://github.com/user-attachments/assets/bfecc526-34c2-4f6c-adba-dfee7ee29611" />

I noticed that when saving the comparison images, the rendered colors appear darker and inconsistent. However, when generating a video from the same data, the colors look normal. How can this discrepancy be explained?Thanks！

**Comment by jianglh-WHU** [comment]
That's very strange. I suggest that you can check the image range, [0-1] or [-1~1 ].

**Comment by SanMumumu** [comment]
### Description

When loading the provided pretrained weights, the rendered outputs differ significantly from the ground truth.

* Using `rgb_pred = output.color[0].float()` produces abnormal colors.
* Using `rgb_pred = (output.color[0].float() + 1) / 2` improves the result slightly, but it is still far from the GT.

### Steps to Reproduce


```python
model_weights = load_file("/mnt/hwdata/wangsen/anysplat_ckpt/model.safetensors")
model.load_state_dict(model_weights, strict=False)
```

https://github.com/InternRobotics/AnySplat/blob/57937e4e29a30ed65e86d8813edd4a4797b0696c/src/model/model_wrapper.py#L486

```python
rgb_pred = output.color[0].float()
# or
rgb_pred = (output.color[0].float() + 1) / 2
```

### Expected Behavior

Rendered output should visually match the GT.

### Actual Behavior

Both raw and normalized outputs deviate strongly from GT.

### Examples

* Raw output:
  ![direct](https://github.com/user-attachments/assets/d8fb5170-a2ec-4552-82d9-330c1b8710a6)

* Normalized `(x+1)/2`:
  ![normalized](https://github.com/user-attachments/assets/617bc951-d24b-4443-b065-2bb802df0a6b)

* Ground truth:
  ![GT](https://github.com/user-attachments/assets/4ed48a73-b95d-45cf-a612-f1ee2fade185)

**Comment by AntaresAlice** [comment]
Noticed the same problem. Has anyone figured out how to solve?

**Comment by SanMumumu** [comment]
@AntaresAlice @uJunLI 

In my implementation, we commented out the line of code at https://github.com/InternRobotics/AnySplat/blob/57937e4e29a30ed65e86d8813edd4a4797b0696c/src/model/model_wrapper.py#L462
, which previously mapped the range of `batch["context"]["image"]` from $[0,1]$ to $[-1,1]$ before feeding it into the VGGT module. Given that VGGT is designed to process input images within the $[0,1]$ range, this normalization step was unnecessary. Empirically, removing it not only preserved correctness but also improved the quality of VGGT’s depth predictions.

Furthermore, two minor modifications were introduced for consistency:

1. At https://github.com/InternRobotics/AnySplat/blob/57937e4e29a30ed65e86d8813edd4a4797b0696c/src/model/model_wrapper.py#L495, we changed  to

   ```python
   rgb_gt = batch["context"]["image"][0].float()
   ```

2. At https://github.com/InternRobotics/AnySplat/blob/57937e4e29a30ed65e86d8813edd4a4797b0696c/src/model/model_wrapper.py#L521, we changed  to

   ```python
   context_img = batch["context"]["image"][0]
   ```

After these modifications, the pipeline behaved as expected, yielding stable and improved depth estimation results.

---
Only for use during validation; if you want to use it in training, remember to modify the loss.

**Comment by terencepp** [comment]
> [@AntaresAlice](https://github.com/AntaresAlice) [@uJunLI](https://github.com/uJunLI)
> 
> In my implementation, we commented out the line of code at
> 
> [AnySplat/src/model/model_wrapper.py](https://github.com/InternRobotics/AnySplat/blob/57937e4e29a30ed65e86d8813edd4a4797b0696c/src/model/model_wrapper.py#L462)
> 
> Line 462 in [57937e4](/InternRobotics/AnySplat/commit/57937e4e29a30ed65e86d8813edd4a4797b0696c)
> 
>  batch: BatchedExample = self.data_shim(batch) 
> 
> , which previously mapped the range of `batch["context"]["image"]` from 
> [
> 0
> ,
> 1
> ]
>  to 
> [
> −
> 1
> ,
> 1
> ]
>  before feeding it into the VGGT module. Given that VGGT is designed to process input images within the 
> [
> 0
> ,
> 1
> ]
>  range, this normalization step was unnecessary. Empirically, removing it not only preserved correctness but also improved the quality of VGGT’s depth predictions.
> Furthermore, two minor modifications were introduced for consistency:
> 
> 1. At 
>      
>        
>          [AnySplat/src/model/model_wrapper.py](https://github.com/InternRobotics/AnySplat/blob/57937e4e29a30ed65e86d8813edd4a4797b0696c/src/model/model_wrapper.py#L495)
>        
>        
>             Line 495
>          in
>          [57937e4](/InternRobotics/AnySplat/commit/57937e4e29a30ed65e86d8813edd4a4797b0696c)
>        
>      
>      
>        
>    
>            
>              
>               rgb_gt = (batch["context"]["image"][0].float() + 1) / 2 
>            
>        
>      
>    
>    , we changed  to
>    rgb_gt = batch["context"]["image"][0].float()
> 2. At 
>      
>        
>          [AnySplat/src/model/model_wrapper.py](https://github.com/InternRobotics/AnySplat/blob/57937e4e29a30ed65e86d8813edd4a4797b0696c/src/model/model_wrapper.py#L521)
>        
>        
>             Line 521
>          in
>          [57937e4](/InternRobotics/AnySplat/commit/57937e4e29a30ed65e86d8813edd4a4797b0696c)
>        
>      
>      
>        
>    
>            
>              
>               context_img = inverse_normalize(batch["context"]["image"][0]) 
>            
>        
>      
>    
>    , we changed  to
>    context_img = batch["context"]["image"][0]
> 
> After these modifications, the pipeline behaved as expected, yielding stable and improved depth estimation results.
> 
> Only for use during validation; if you want to use it in training, remember to modify the loss.

Hello, thank you for your useful modification.

could you tell me why i cannot do this modification during training?

**Comment by LiuChunlin-eng** [comment]
> > To be honest, I’m not sure. I haven’t conducted experiments where the model is overfitted on a small subset. You might need to try using a larger one.
> 
> <img alt="Image" width="1853" height="960" src="https://private-user-images.githubusercontent.com/143502956/479779693-bfecc526-34c2-4f6c-adba-dfee7ee29611.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjY4MzUwOTgsIm5iZiI6MTc2NjgzNDc5OCwicGF0aCI6Ii8xNDM1MDI5NTYvNDc5Nzc5NjkzLWJmZWNjNTI2LTM0YzItNGY2Yy1hZGJhLWRmZWU3ZWUyOTYxMS5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIyN1QxMTI2MzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kNTFkNzczM2UyZmJmNTM1NWIyYTUxNGY4YmY2ZTkxMTlhYjRmYTI1MDBlYjQ5ZjFhZjhiYTNmYjcwMTljNDQ4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.cJFzV0CKHjLlFDt7Ma7rvv_GWo8r0XFvPPKoB35x4h8">
> I noticed that when saving the comparison images, the rendered colors appear darker and inconsistent. However, when generating a video from the same data, the colors look normal. How can this discrepancy be explained?Thanks！

I guess you are training this from scratch. Please note that when training Gaussian parameters from scratch, they are random and haven't started learning yet, which is why the colors don't match the GT.
